onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -132,42 +132,34 @@ def data(self, request, **kwargs):
                 qs_kwargs.update({'id__gt': gt_id})
 
             instances = Instance.objects.filter(**qs_kwargs).order_by('pk')
-            length = self.total_count = instances.count()
+            self.total_count = instances.count()
```

Comment:
this should be in-lined into the `count` conditinoal on line 138 so we don't calculate it by default